File: backend/schemas/order_attachment.py
"""
Pydantic schemas for OrderAttachment models - WideOrbit-compatible
"""

from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime
from uuid import UUID
from backend.models.order_attachment import AttachmentType


class OrderAttachmentBase(BaseModel):
    """Base order attachment schema"""
    file_name: str
    mime_type: Optional[str] = None
    file_size: Optional[int] = None
    attachment_type: AttachmentType
    description: Optional[str] = None
    notes: Optional[str] = None


class OrderAttachmentCreate(OrderAttachmentBase):
    """Schema for creating a new order attachment"""
    order_id: Optional[UUID] = None
    order_line_id: Optional[UUID] = None
    copy_id: Optional[UUID] = None
    file_path: str


class OrderAttachmentUpdate(BaseModel):
    """Schema for updating an order attachment"""
    file_name: Optional[str] = None
    mime_type: Optional[str] = None
    attachment_type: Optional[AttachmentType] = None
    description: Optional[str] = None
    notes: Optional[str] = None


class OrderAttachmentResponse(OrderAttachmentBase):
    """Schema for order attachment response"""
    id: UUID
    order_id: Optional[UUID] = None
    order_line_id: Optional[UUID] = None
    copy_id: Optional[UUID] = None
    file_path: str
    uploaded_by: UUID
    uploaded_by_name: Optional[str] = None
    uploaded_at: datetime
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: archive/librelog-web-api-original/src/main/java/com/onelpro/librelog/models/OrderAttachment.java
package com.onelpro.librelog.models;

import com.onelpro.librelog.enums.AttachmentType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

/**
 * Entity representing an order attachment in the system.
 */
@Entity
@Table(name = "order_attachments", indexes = {
        @Index(name = "ix_order_attachments_order_id", columnList = "order_id"),
        @Index(name = "ix_order_attachments_order_line_id", columnList = "order_line_id"),
        @Index(name = "ix_order_attachments_copy_id", columnList = "copy_id"),
        @Index(name = "ix_order_attachments_attachment_type", columnList = "attachment_type"),
        @Index(name = "ix_order_attachments_uploaded_at", columnList = "uploaded_at")
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class OrderAttachment {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id")
    private Order order;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_line_id")
    private OrderLine orderLine;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "copy_id")
    private Copy copy;

    @Column(name = "file_path", nullable = false, columnDefinition = "TEXT")
    private String filePath;

    @Column(name = "file_name", nullable = false, length = 255)
    private String fileName;

    @Column(name = "mime_type", length = 100)
    private String mimeType;

    @Column(name = "file_size")
    private Long fileSize;

    @Enumerated(EnumType.STRING)
    @Column(name = "attachment_type", nullable = false, columnDefinition = "attachmenttype")
    private AttachmentType attachmentType;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "notes", columnDefinition = "TEXT")
    private String notes;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "uploaded_by", nullable = false)
    private User uploadedBy;

    @Column(name = "uploaded_at")
    private Instant uploadedAt;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;
}

